src/modules/container_registries/private_endpoint.tf

```diff
@@ -0,0 +1,17 @@
+resource "azurerm_private_endpoint" "main" {
+  name                = var.settings.private_endpoint.name
+  resource_group_name = local.resource_group_name
+  location            = local.location
+  subnet_id           = local.subnet_id
+
+  tags = local.tags
+
+  private_service_connection {
+    name                           = var.settings.private_endpoint.private_service_connection.name
```

Comment:
The private service name should be optional and defaulting to `psc-${var.settings.name}` for all resources that may define a private endpoint.